src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/ExecuteRequests/ExecuteAndReturnResultRequest.cs

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts.ExecuteRequests
+{
+    /// <summary>
+    /// Parameters for executing a query from a document open in the workspace
+    /// </summary>
+    public class ExecuteAndReturnResultParams
+    {
+        /// <summary>
+        /// The string to execute
+        /// </summary>
+        public string QueryString { get; set; }
+
+        /// <summary>
+        /// The owneruri to get connection from
+        /// </summary>
+        public string OwnerUri { get; set; }
+    }
+
+    /// <summary>
+    /// Result
+    /// </summary>
+    public class ExecuteAndReturnResultResult
+    {
+
+        /// <summary>
+        /// The number of rows that was returned with the resultset
+        /// </summary>
+        public long RowCount { get; set; }
+
+        /// <summary>
+        /// Details about the columns that are provided as solutions
+        /// </summary>
+        public DbColumnWrapper[] ColumnInfo { get; set; }
+
+        /// <summary>
+        /// 2D array of the cell values requested from result set
+        /// </summary>
+        public DbCellValue[][] Rows { get; set; }
+    }
+
+    public class ExecuteAndReturnResultRequest
+    {
+        public static readonly
+            RequestType<ExecuteAndReturnResultParams, ExecuteAndReturnResultResult> Type =

```

Comment:
The asynchronous of this request is given by the asynchronous of the entire query execution (see **https://github.com/Microsoft/sqltoolsservice/blob/master/src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs#L565**). I don't believe the message channel ever waits for a message. I believe the synchronous problem we were having before was because the handlers were not asynchronous, not because the message channel was waiting on a response. Having a separate notification complete pretty much defeats the purpose of having a simple "execute this query" and we may as well just use the query string execution message.